torch/distributed/_tensor/api.py

```diff
@@ -362,10 +374,14 @@ def from_local(
             run_check,
             shape,
             stride,
+            async_output,
         )
 
     def to_local(
-        self, *, grad_placements: Optional[Sequence[Placement]] = None
+        self,
+        *,
+        grad_placements: Optional[Sequence[Placement]] = None,
+        async_output: str = True,
```

Comment:
What's the concrete use case of having this exposed as a API, instead of doing something like below explicitly:

````
out = dtensor.to_local()
if isinstance(out, AsyncCollectiveTensor):
    out = out.wait()
```
We should be restrictive to what public API we expose so unless there's a strong reason to do this I would propose we do this manually

Reply:
Thanks for the suggestion. There're are two major places in the DTensor API where `to_local()` is called, one is the forward `DTensor.to_local()` and the other is the `DTensor.from_local()` backward. For the second one, any recommend way to do the check and wait() (i.e. in the backward hook)?